Neos.Fusion/Classes/Core/ParserInterface.php

```diff
@@ -22,6 +22,17 @@
  */
 interface ParserInterface
 {
+    /**
+     * Parses the given Fusion source code, resolves includes and returns a merged array tree
+     * as the result.
+     *
+     * @param FusionSourceCode|FusionSourceCodeCollection $sourceCode The Fusion source code to parse
+     * @return array The merged array tree for the Fusion runtime, generated from the source code
+     * @throws Fusion\Exception
+     * @api
+     */
+    public function parseFrom(FusionSourceCode|FusionSourceCodeCollection $sourceCode): array;
```

Comment:
This makes it a breaking change, strictly speaking.

Reply:
because a new method?

@mficzel and me decided to create a new method than to make the old `parse` method understand `FusionSourceCodeCollection` via union types as this makes things uneccessary complex

i think this is fine

Reply:
Simply because it changes an interface marked as `@api` 😉

Reply:
ahh i see xD

is this really important? otherwise i would add it to the interface later ... (with 9) -> or forget it ^^

Reply:
Well… "strictly speaking". The rules would say, we must put a breaking change in the next major. The again, the `ParserInterface` is something I was surprised to see as `@api` and I really doubt someone already implemented their own. 🙃

If noone else objects, fine with me!

Reply:
yes the `@api` seems odd -> as it cannot even be changed via Objects yaml as we use `new Parser()`

Reply:
Then maybe drop it, before someone notices? 🙈

Reply:
^^
about removing the interface @mficzel said: https://github.com/neos/neos-development-collection/pull/3497#issuecomment-974801376

Reply:
True… then again, `Parser` has only one public method, so that makes it clear, too. But of course we can leave the interface in and just drop the `@api`.

Reply:
Fine with me either way. Was my gut feeling at first too.

Maybe @kitsunet can help us out?

Reply:
> Well… "strictly speaking". The rules would say, we must put a breaking change in the next major. The again, the `ParserInterface` is something I was surprised to see as `@api` and I really doubt someone already implemented their own. 🙃
> 
> If noone else objects, fine with me!

I'd say `@api` or not, it's an interface, which _could_ strictly speaking be replaced. Now I doubt anyone did, but that falls under famous last words. Do we need this change for anything really important like NOW? Otherwise I would prefer next major.

Reply:
Nope the `ParserInterface` **CANNOT** be replaced via objects.yaml. Its only usage/reference is in beeing implemented by the `Parser`. It has **no use** than to delay us in the discussion ^^

We have 3 Solutions:

- add the new method to the `ParserInterface` in 9.0 (drop the `ParserInterface` in 9.0)
- drop the  `ParserInterface` now - as it has no use - and 0 breakiness - https://github.com/neos/neos-development-collection/pull/3839#discussion_r991112012
- add the new method to the `ParserInterface` now


im team `drop the  `ParserInterface` now`

any objections?

Reply:
Given that it's a useless interface I doubt anyone has really implemented it, so from my side I would be fine with this technically breaking change. I am SURE there is a project where someone imlpemented the interface but never checked if it works and it breaks, but weeeeeell......

Reply:
just to be 100% clear this goes into 8.0

Reply:
I got lost in the little things - this is not what this PR should focus on - nothing will be changed about the `ParserInterface`